### ISSUE TITLE
Consider distributing wheels on PyPI

### ISSUE THREAD

**Issue by matthewfeickert** [gammapy/gammapy#4656]
**Is your feature request related to a problem? Please describe.**

At the moment there are no wheels on PyPI for `gammapy` (e.g. c.f. https://pypi.org/project/gammapy/1.1/#files for `gammapy` `v1.1`). As `gammapy` is pure-Python it should be straightforward to build wheels along with the `sdist` with `build`. You're already using `build` though in

https://github.com/gammapy/gammapy/blob/d775861ff491f1a0301d5505aa039f24eb3402bc/.github/workflows/release.yml#L21-L25

so is there a reason why you don't build and distribute wheels?

**Describe the solution you'd like**

Build and distribute `wheels` to PyPI (can use a workflow like https://github.com/scikit-hep/pyhf/blob/4845a154c18edfd5e2b29ef94167b2818fa6e674/.github/workflows/publish-package.yml).

**Describe alternatives you've considered**

There is no alternative, but there might be a reason that you intentionally don't distribute wheels that I'm missing.

**Additional context**
N/A

**Comment by adonath** [comment]
Thanks @matthewfeickert, we should absolutely do this!

**Comment by adonath** [comment]
I think we should prioritize this. I am currently working on a machine without root access and if the `python-dev` packages are missing (`Python.h` etc.) it is very troublesome to install Gammapy because of the Cython extensions. Alternatively we should consider making the Cython extensions optional and have the pure Numpy fallback.

**Comment by maxnoe** [comment]
I can open a PR with a basic cibuildwheel config

**Comment by bkhelifi** [comment]
Thanks!

I have just discovered that one can add the licence in the wheel distribution (see https://pythonwheels.com/) by adding 
```
[metadata]
license_files = LICENSE
```
Probably a detail to be considered

**Comment by maxnoe** [comment]
This is already setup: https://github.com/gammapy/gammapy/blob/520c1348e7fe3b94a99aabdac216e03450cb4f7d/setup.cfg#L7

**Comment by maxnoe** [comment]
>  it is very troublesome to install Gammapy because of the Cython extensions.

Is having to install micromamba and creating an env really "very troublesome", or is there a specific requirement for the usecase of using the system python without having root access and dev packages not being available?

Adding the cibuildwheel step was easy though, relying on the openastronomy actions definition.

**Comment by adonath** [comment]
Thanks @maxnoe! In my case I wanted to install Gammapy alongside PyTorch and PyTorch does not ship conda builds anymore. When pip installing into the conda env I ran into an incompatibility issue, so I tried installing both into a venv and ran into the issue above. This all happened on a cluster, where I do not have root access. I think installing PyTorch and Gammapy in the same environment would be good to support.

**Comment by matthewfeickert** [comment]
> PyTorch does not ship conda builds anymore

I feel like this was very poorly communicated by the PyTorch team. The `pytorch` conda channel will no longer be updated but PyTorch will still be distributed **on conda-forge**. While true that this will be "community supported" the same is true for basically anything on conda-forge, and "the community" in this case still includes PyTorch team members.

But I would saying having wheels on PyPI and distributions on conda-forge (which you already have) would be a big win. [`cibuildwheel`](https://cibuildwheel.pypa.io/) is your friend here for dealing with wheel building.

**Comment by maxnoe** [comment]
> But I would saying having wheels on PyPI and distributions on conda-forge (which you already have) would be a big win. [cibuildwheel](https://cibuildwheel.pypa.io/) is your friend here for dealing with wheel building.

See https://github.com/gammapy/gammapy/pull/5708

**Comment by adonath** [comment]
> I feel like this was very poorly communicated by the PyTorch team. The pytorch conda channel will no longer be updated but PyTorch will still be distributed on conda-forge

Yes, especially because it is not mentioned in the official docs. Where the install matrix for conda just shows:

![Image](https://github.com/user-attachments/assets/535dd40d-447d-486e-8f4e-9613f88e7236)

**Comment by matthewfeickert** [comment]
Thanks!
